File: longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0,n=s.length();
        for(int i=0;i<n;i++)
        {
            int count[10000] = {0};
            int len = 1;
            count[s.at(i)]++;
            ans = max(ans,len);
            for(int j=i+1;j<n;j++)
            {
                count[s.at(j)]++;
                if(count[s.at(j)] > 1)
                    break;
                len++;
                ans = max(ans,len);
            }
        }
        return ans;
    }
};

File: January LeetCoding Challenge/Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] lastOcurrance = new int[300];
        for(int i=0;i<300;i++)lastOcurrance[i] = -1;
        char[] ch = s.toCharArray();
        int max = 0,len = 0;
        for(int i=0;i<ch.length;i++)
        {
            int dis = i - lastOcurrance[ch[i]];
            if(dis <= len)len = dis;
            else len++;
            max = Math.max(max,len);
            lastOcurrance[ch[i]] = i;
        }
        return max;
    }
}
